### PR TITLE
Initial transaction trace added

### PR DESCRIPTION
**Pull request by coop**
```
root@9b9a03dc0711:/# nginx -V
nginx version: nginx/1.4.6 (Ubuntu)
built by gcc 4.8.2 (Ubuntu 4.8.2-19ubuntu1)
TLS SNI support enabled
configure arguments: --with-cc-opt='-g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2' --with-ld-opt='-Wl,-Bsymbolic-functions -Wl,-z,relro' --prefix=/usr/share/nginx --conf-path=/etc/nginx/nginx.conf --http-log-path=/var/log/nginx/access.log --error-log-path=/var/log/nginx/error.log --lock-path=/var/lock/nginx.lock --pid-path=/run/nginx.pid --http-client-body-temp-path=/var/lib/nginx/body --http-fastcgi-temp-path=/var/lib/nginx/fastcgi --http-proxy-temp-path=/var/lib/nginx/proxy --http-scgi-temp-path=/var/lib/nginx/scgi --http-uwsgi-temp-path=/var/lib/nginx/uwsgi --with-debug --with-pcre-jit --with-ipv6 --with-http_ssl_module --with-http_stub_status_module --with-http_realip_module --with-http_addition_module --with-http_dav_module --with-http_geoip_module --with-http_gzip_static_module --with-http_image_filter_module --with-http_spdy_module --with-http_sub_module --with-http_xslt_module --with-mail --with-mail_ssl_module --add-module=/nginx-1.4.6/debian/modules/nginx-auth-pam --add-module=/nginx-1.4.6/debian/modules/nginx-dav-ext-module --add-module=/nginx-1.4.6/debian/modules/nginx-echo --add-module=/nginx-1.4.6/debian/modules/nginx-upstream-fair --add-module=/nginx-1.4.6/debian/modules/ngx_http_substitutions_filter_module --add-module=/nginx-1.4.6/debian/modules/ngx_txid-master

root@9b9a03dc0711:/# cat /etc/nginx/sites-enabled/default
server {
  listen 80 default_server;
  listen [::]:80 default_server ipv6only=on;

  root /usr/share/nginx/html;
  index index.html index.htm;

  # Make site accessible from http://localhost/
  server_name localhost;

  location / {
    add_header       X-Request-Id $txid;
    proxy_set_header X-Request-Id $txid;

    # First attempt to serve request as file, then
    # as directory, then fall back to displaying a 404.
    try_files $uri $uri/ =404;
    # Uncomment to enable naxsi on this location
    # include /etc/nginx/naxsi.rules
  }
}

root@9b9a03dc0711:/# curl -I localhost
HTTP/1.1 200 OK
Server: nginx/1.4.6 (Ubuntu)
Date: Mon, 02 Mar 2015 08:34:35 GMT
Content-Type: text/html
Content-Length: 612
Last-Modified: Tue, 04 Mar 2014 11:46:45 GMT
Connection: keep-alive
ETag: "5315bd25-264"
X-Request-Id: abr6flvlp9ukdi15gbmg
Accept-Ranges: bytes
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
-RUN apt-get update && apt-get install -y  \
-        geoip-database \
-        libgeoip1 \
-        nginx-full \
-    && rm -rf /var/lib/apt/lists/*
+WORKDIR /
+
+ADD https://gist.githubusercontent.com/coop/922941b03a0c2af631a7/raw/935b474edfc77636156a8b4f3c7c7b4d31ad40c1/rules.patch /
+
+RUN apt-get update && \
```
Comment (coop) [inline Dockerfile:8]: I'll wrap these up into a single run command and then it should be ready to go.

**Comment by evilmarty** [comment]
Is this to add the request id?

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+    apt-get install -y  \
+    geoip-database \
+    libgeoip1 \
+    wget && \
+    apt-get build-dep nginx-full -y && \
+    apt-get source nginx-full
+
+RUN $(cd nginx-1.4.6/debian/modules && wget https://github.com/streadway/ngx_txid/archive/master.tar.gz -O - | tar -xz) && \
```
Comment (justinhennessy) [inline Dockerfile:16]: Should we be hard coding the version in here? wouldn't it make more send to specify the version dynamically (though I see the rules patch has it as well.

**Comment by dansowter** [comment]
Covered in https://github.com/everydayhero/nginx-docker/pull/3

**Pull request closed without merging**